src/ORM/Table.php

```diff
@@ -1513,18 +1513,20 @@ public function save(EntityInterface $entity, $options = [])
     }
 
     /**
-     * Try to save an entity and fail if save was not successful
+     * Try to save an entity or throw a PersistenceFailedException if the application rules checks failed,
+     * the entity contains errors or the save was aborted by a callback.
      *
      * @param \Cake\Datasource\EntityInterface $entity the entity to be saved
      * @param array|\ArrayAccess $options The options to use when saving.
      * @return \Cake\Datasource\EntityInterface
-     * @throws \RuntimeException When the entity couldn't be saved
+     * @throws \PersistenceFailedException When the entity couldn't be saved
+     * @see \Cake\ORM\Table::save()
      */
     public function saveOrFail(EntityInterface $entity, $options = [])
     {
         $saved = $this->save($entity, $options);
         if ($saved === false) {
-            throw new PersistenceFailedException(['save', $entity]);
+            throw new PersistenceFailedException(['save', $entity->source(), $entity->{$this->primaryKey()}, print_r($entity->errors(), true)]);
```

Comment:
@robertpustulka Which values should I extract from the entity? Can you give me an example how the message should look like in the end? Also, is there any Exception class in cake that does sth. similar so I can get a first impression on how to code this?